hw/ip/usbdev/data/usbdev_testplan.hjson

```diff
@@ -18,12 +19,824 @@
+
+            - Enable the interrupt for RX full by setting INTR_ENABLE[8] and then send back to back packets
+              to the device.
+            - Don't just service the packets received by popping out from rx_fifo.
+            - Verify that this interrupt pin should go high once the received fifo is full.
+            '''
+      stage: V1
+      tests: ["usbdev_rx_full"]
+    }
+    {
+      name: av_overflow
+      desc: '''
+            Verify that if a write was done to the Available Buffer FIFO when the FIFO was full then
+            av_overflow signal is high.
+
+            - Write a particular buffer ID to the available buffer FIFO when it is already full.
+            - Verify that this interrupt pin goes high indicating that the available fifo has
+              reached the overflowing condition.
+            '''
+      stage: V1
+      tests: ["usbdev_av_overflow"]
+    }
+    {
+      name: enable
+      desc: '''
+            Verify that USB connect to interface when enable is set.
+
+            - set USB_CTRL_enable pin high.
+            - Make sure phy_config_pinflip isn't set.
+            - Verify that the usb_dp_pullup_o pin goes high.
+            '''
+      stage: V1
+      tests: ["usbdev_enable"]
+    }
+    {
+      name: resume_link_active
+      desc: '''
+            Verify that when Write a 1 to this bit to instruct usbdev to jump to the LinkResuming state
+            then write will only have an effect when the device is in the LinkPowered state.
+
+            - Perform no operation in J state for more than 3 ms.
+            - Read the link state from usbstate register to see if it is in powered state and then send the
+              resume signal (K state) for 20 ms and set the usbctrl.resume_link_active.
+            - Verify that the resume_link_active interrupt goes high and device will be resumed.
+            '''
+      stage: V1
+      tests: ["usbdev_resume_link_active"]
+    }
+    {
+      name: device_address
+      desc: '''
+            Verify that device address set by host (this should be copied from the Set Device ID SETUP packet).
+            This will be zeroed by the hardware when the link resets.
+
+            - read USB_CTRL[22:16] send an In token packet to the device on an enabled endpoint with read value
+              as the device address.
+            - Verify that if Device send any data packet in response then the read value is the actual address of the device.
+            '''
+      stage: V1
+      tests: ["usbdev_device_address"]
+    }
+    {
+      name: link_in_err
+      desc: '''
+            Verify that if a packet to an IN endpoint started to be received but was then dropped due to an error
+            then link_in_err will raise.
+
+            - send an IN token to the device for initiating an IN transaction.
+            - The device should respond with IN data packets.
+            - In the handshake send an invalid CRC or a different PID then expected.
```

Comment:
It is up to the DV environment, in simulating a host controller, to reject IN packets in this test. It may return an unexpected PID in the handshake packet, or act *as if* the CRC16 mismatched against the received data.